apps/web/src/components/NavMenusSection/MobileMenu.tsx

```diff
@@ -0,0 +1,159 @@
+import { useEffect, useState } from "react";
+import { usePathname } from "next/navigation";
+import { ChevronDownIcon } from "@heroicons/react/24/outline";
+import { Bars3Icon } from "@heroicons/react/24/solid";
+
+import { BlobscanLogo } from "../BlobscanLogo";
+import { Button } from "../Button";
+import { Collapsable } from "../Collapsable";
+import { Rotable } from "../Rotable";
+import { ThemeModeButton } from "../ThemeModeButton";
+import type { ExpandibleMenuItem, MenuItem } from "./data";
+import { MENU_DATA } from "./data";
+
+export function MobileMenu() {
+  const [open, setOpen] = useState(false);
+
+  useEffect(() => {
+    if (open) {
+      document.body.style.overflow = "hidden";
+    } else {
+      document.body.style.overflow = "auto";
+    }
+  }, [open]);
+
+  return (
+    <>
+      <Button
+        variant="icon"
+        className="md:hidden"
```

Comment:
```suggestion
```

I prefer to delegate the responsibility of hiding or displaying a component to the component’s consumer.

Reply:
When using `MobileMenu` we'll wrap it within a `div` and add this responsive tailwind classes

Reply:
Done [d0a163d](https://github.com/Blobscan/blobscan/pull/469/commits/d0a163dbdb8131bc4286b926ac30f8c536bf521a)